state-chain/runtime/src/chainflip.rs

```diff
@@ -354,31 +360,20 @@ impl TransactionBuilder<Solana, SolanaApi<SolEnvironment>> for SolanaTransaction
 		// to modify the apicall, by replacing the aggkey with the new key, in the key_accounts in
 		// the tx's message to create a new valid threshold signing payload.
 		maybe_current_onchain_key.map_or(RequiresSignatureRefresh::False, |current_on_chain_key| {
-			match call.signer().and_then(|signer| {
-				if signer != current_on_chain_key {
-					Some(signer)
-				} else {
-					None
-				}
-			}) {
-				Some(signer) => {
+			match call.signer() {
+				Some(signer) if signer != current_on_chain_key => {
 					let mut modified_call = (*call).clone();
-					// the unwraps should be safe because we are in the code where on chain key
-					// already exists (see above) and so thecurrent_key_epoch should also exist and
-					// so should the corresponding key,
-					let current_aggkey = SolanaThresholdSigner::keys(
-						SolanaThresholdSigner::current_key_epoch().unwrap(),
-					)
-					.unwrap();
-					for (i, key) in call.transaction.message.account_keys.iter().enumerate() {
+					// the unwrap should be safe because we are in the code where on chain key
+					// already exists (see above) and so the active_epoch_key() should also exist.
+					let current_aggkey = SolanaThresholdSigner::active_epoch_key().unwrap().key;
```

Comment:
I know this is safe *now* but we can't guarantee it will always be safe (maybe someone makes a change that temporarily removes keys, or we add a feature for deactivating keys that causes this call to return None... ) . I would prefer if we log an error and return `RequiresSignatureRefresh::False` rather than using unwrap.

Reply:
(Apart from that, lgtm)